### PR TITLE
GAIA: Include the parameter linking_parameter in the function get_datalinks

### PR DESCRIPTION
**Pull request by cosmoJFH**
In the PR [#2859] the new parameter ``linking_parameter`` was introduced in the method ``load_data`` .

The same parameter should be available in the the function ``get_datalinks``, since in the next realeas of the gaia archive (February/March) the ids will be identified by this new parameter.

This PR contains new tests. We have also reviewed and updated the docstrings of the functions.

cc @esdc-esac-esa-int

### CODE REVIEW

File: astroquery/gaia/core.py [hunk 1]
```diff
@@ -569,7 +627,7 @@ def __cone_search(self, coordinate, radius, *, table_name=None,
             when the job is executed in asynchronous mode, this flag specifies
             whether the execution will wait until results are available
         output_file : str, optional, default None
-            file name where the results are saved if dump_to_file is True.
+            file name where the results are saved if `dump_to_file` is True.
```
Comment (bsipocz) [inline astroquery/gaia/core.py:630]: sphinx tries to resolve the single backticks into links, and there is no API docs anchor for them, so all these needs to be in double backticks to make the docs build pass. I'm sorry about that, I'll add fixes as a commit, so this won't hold up a PR.

```suggestion
            file name where the results are saved if ``dump_to_file`` is True.
```

File: astroquery/gaia/core.py [hunk 2]
```diff
@@ -647,17 +705,17 @@ def cone_search(self, coordinate, *, radius=None,
...
+        radius : str or astropy.units, mandatory
             radius
         table_name : str, optional, default main gaia table name doing the cone search against
         ra_column_name : str, optional, default ra column in main gaia table
             ra column doing the cone search against
         dec_column_name : str, optional, default dec column in main gaia table
             dec column doing the cone search against
         output_file : str, optional, default None
-            file name where the results are saved if dump_to_file is True.
+            file name where the results are saved if `dump_to_file` is True.
```
Comment (bsipocz) [inline astroquery/gaia/core.py:718]:
```suggestion
            file name where the results are saved if ``dump_to_file`` is True.
```

File: astroquery/gaia/core.py [hunk 3]
```diff
@@ -711,7 +769,7 @@ def cone_search_async(self, coordinate, *, radius=None,
             specifies whether
             the execution will wait until results are available
         output_file : str, optional, default None
-            file name where the results are saved if dump_to_file is True.
+            file name where the results are saved if `dump_to_file` is True.
```
Comment (bsipocz) [inline astroquery/gaia/core.py:772]:
```suggestion
            file name where the results are saved if ``dump_to_file`` is True.
```

File: astroquery/gaia/core.py [hunk 4]
```diff
@@ -772,6 +830,15 @@ def __getCoordInput(self, value, msg):
 
     @staticmethod
     def correct_table_units(table):
+        """Correct format in the units of the columns
+        TAP & TAP+
+
+        Parameters
+        ----------
+        table :Table, mandatory
```
Comment (bsipocz) [inline astroquery/gaia/core.py:838]: sphinx needs spaces on both sides of the `:` when listing parameters (this won't cause a warning/error, just won't render properly)
```suggestion
        table : `~astropy.table.Table`, mandatory
```

File: astroquery/gaia/core.py [hunk 5]
```diff
@@ -877,7 +944,7 @@ def launch_job(self, query, *, name=None, output_file=None,
         name : str, optional, default None
             custom name defined by the user for the job that is going to be created
         output_file : str, optional, default None
-            file name where the results are saved if dump_to_file is True.
+            file name where the results are saved if `dump_to_file` is True.
```
Comment (bsipocz) [inline astroquery/gaia/core.py:947]:
```suggestion
            file name where the results are saved if ``dump_to_file`` is True.
```

File: astroquery/gaia/core.py [hunk 6]
```diff
@@ -892,7 +959,7 @@ def launch_job(self, query, *, name=None, output_file=None,
             resource to be uploaded to UPLOAD_SCHEMA
         upload_table_name : str, optional, default None
             resource temporary table name associated to the uploaded resource.
-            This argument is required if upload_resource is provided.
+            This argument is required if `upload_resource` is provided.
```
Comment (bsipocz) [inline astroquery/gaia/core.py:962]:
```suggestion
            This argument is required if ``upload_resource`` is provided.
```

File: astroquery/gaia/core.py [hunk 7]
```diff
@@ -922,7 +989,7 @@ def launch_job_async(self, query, *, name=None, output_file=None,
         name : str, optional, default None
             custom name defined by the user for the job that is going to be created
         output_file : str, optional, default None
-            file name where the results are saved if dump_to_file is True.
+            file name where the results are saved if `dump_to_file` is True.
```
Comment (bsipocz) [inline astroquery/gaia/core.py:992]:
```suggestion
            file name where the results are saved if ``dump_to_file`` is True.
```

File: astroquery/gaia/core.py [hunk 8]
```diff
@@ -940,7 +1007,7 @@ def launch_job_async(self, query, *, name=None, output_file=None,
             resource to be uploaded to UPLOAD_SCHEMA
         upload_table_name : str, optional, default None
             resource temporary table name associated to the uploaded resource.
-            This argument is required if upload_resource is provided.
+            This argument is required if `upload_resource` is provided.
```
Comment (bsipocz) [inline astroquery/gaia/core.py:1010]:
```suggestion
            This argument is required if ``upload_resource`` is provided.
```

**Review by bsipocz — APPROVED** [review]
Looks all good, thank you @cosmoJFH!
I'll commit the fixes for sphinx, as well the unrelated fix for a failing doctest and this will be ready to go in.

**Comment by bsipocz** [comment]
Thanks @cosmoJFH!

**Comment by cosmoJFH** [comment]
Thanks @bsipocz for reviewing the PR!

**Pull request merged**